gammapy/modeling/models/temporal.py

```diff
@@ -55,6 +61,22 @@ def __call__(self, time):
     def type(self):
         return self._type
 
+    @property
+    def reference_time(self):
+        """Reference time in mjd"""
+        return Time(self.t_ref.value, format="mjd", scale=self.scale)
+
+    @reference_time.setter
+    def reference_time(self, t_ref):
+        """Reference time"""
+        self.reference_time = getattr(t_ref, self.scale, "mjd")
```

Comment:
Does this work?
I think you should add a check that this is a `Time` object. Then set `t_ref = reference_time.mjd` and `self.scale = t_ref.scale`

Reply:
Should the scale of the model change according to the `Time` passed, or the other way round.
In `SpatialModels`, its the latter, see https://github.com/gammapy/gammapy/blob/610e9d4202b8bef70e61e83a8e73b5b4ca6422b8/gammapy/modeling/models/spatial.py#L104

Reply:
Yes we should follow the same convention. It is still possible to change manually the `scale`.

Reply:
Added a test